python/src/aiconfig/model_parser.py

```diff
@@ -174,11 +174,13 @@ def get_model_settings(self, prompt: Prompt, aiconfig: "AIConfigRuntime") -> Dic
         else:
             # Merge config and prompt settings with prompt settings taking precedent
             model_settings = {}
-            global_settings = aiconfig.get_global_settings(model_metadata.name)
-            prompt_setings = prompt.metadata.model.settings if prompt.metadata.model.settings is not None else {}
+            prompt_settings = prompt.metadata.model.settings if prompt.metadata.model.settings is not None else {}
+
+            model_name = prompt_settings.get("model", model_metadata.name)
+            global_settings = aiconfig.get_global_settings(model_name)
```

Comment:
Hmm, is this required as part of the change? I don't think there's any requirement (currently) that the `model` in the arbitrary model settings has any correlation with global metadata, unlike the prompt metadata model

Reply:
is this the only blocking change requested?

Reply:
I can make this change in a separate diff, but the motivation for this change was this example aiconfig:

```
{
  "name": "NYC Trip Planner",
  "description": "Intrepid explorer with ChatGPT and AIConfig",
  "schema_version": "latest",
  "metadata": {
    "models": {
      "mistralai/Mixtral-8x7B-Instruct-v0.1": {
        "top_p": 1,
        "temperature": 1
        "max_tokens": 3000,
        "system_prompt": "You are an expert travel coordinator with exquisite taste."
      }
    },
    "default_model": "meta-llama/Llama-2-7b-chat-hf"
  },
  "prompts": [
    {
      "name": "gen_itinerary",
      "input": "Generate an itinerary ordered by {{order_by}} for some activities.",
      "metadata": {
        "model": {
          "name": "AnyscaleEndpoint",
          "settings": {
            "model": "mistralai/Mixtral-8x7B-Instruct-v0.1",
            "max_tokens": 3000,
            "system_prompt": "You are an expert travel coordinator with exquisite taste."
          }
        },
        "parameters": {
          "order_by": "geographic location"
        }
      }
    }
  ]
}
```

In this example, we won't be able to read any of the global model settings for the `gen_itinerary` prompt.

Alternative way of expressing the same config would be:
```
{
  "name": "NYC Trip Planner",
  "description": "Intrepid explorer with ChatGPT and AIConfig",
  "schema_version": "latest",
  "metadata": {
    "model_parsers": {
      "mistralai/Mixtral-8x7B-Instruct-v0.1": "AnyscaleEndpoint"
    },
    "models": {
      "mistralai/Mixtral-8x7B-Instruct-v0.1": {
        "top_p": 1,
        "temperature": 1
        "max_tokens": 3000,
        "system_prompt": "You are an expert travel coordinator with exquisite taste."
      }
    },
    "default_model": "meta-llama/Llama-2-7b-chat-hf"
  },
  "prompts": [
    {
      "name": "gen_itinerary",
      "input": "Generate an itinerary ordered by {{order_by}} for some activities.",
      "metadata": {
        "model": {
          "name": "mistralai/Mixtral-8x7B-Instruct-v0.1",
          "settings": {
            "max_tokens": 3000,
            "system_prompt": "You are an expert travel coordinator with exquisite taste."
          }
        },
        "parameters": {
          "order_by": "geographic location"
        }
      }
    }
  ]
}
```

If we are fine with supporting the latter and not the former, then we can keep this code simple and as it was before.

Thoughts @rholinshead @Ankush-lastmile?

Reply:
IMO the latter is correct and the former I actually wouldn't expect to work since the AnyscaleEndpoint 'model' name doesn't match the global settings model -- it's essentially like using a model that hasn't been registered, right?